ui/src/components/ClientsTable/index.jsx

```diff
@@ -95,17 +102,19 @@ export default class ClientsTable extends Component {
         onPageChange={onPageChange}
         renderRow={({ node: client }) => (
           <TableRow key={client.clientId}>
-            <TableCell padding="dense">
-              <TableCellItem
-                dense
-                button
-                component={Link}
-                to={`/auth/clients/${encodeURIComponent(client.clientId)}`}>
-                <ListItemText primary={client.clientId} />
-                <LinkIcon size={iconSize} />
-              </TableCellItem>
+            <TableCell size="small">
+              <Link to={`/auth/clients/${encodeURIComponent(client.clientId)}`}>
+                <TableCellItem button>
+                  <ListItemText>
+                    <Typography className={classes.tableText}>
+                      {client.clientId}
+                    </Typography>
+                  </ListItemText>
+                  <LinkIcon size={iconSize} />
+                </TableCellItem>
+              </Link>
```

Comment:
@helfi92 , Am I doing it right? I want to get rid of refs warnings

Reply:
I think that's the right approach. To get rid of the warnings, I think something like:

```diff
diff --git a/ui/src/components/Dashboard/DocsSidebarList.jsx b/ui/src/components/Dashboard/DocsSidebarList.jsx
index 7055e4cbe..b2273dd33 100644
--- a/ui/src/components/Dashboard/DocsSidebarList.jsx
+++ b/ui/src/components/Dashboard/DocsSidebarList.jsx
@@ -208,17 +208,17 @@ export default class DocsSidebarList extends Component {
       return (
         <Fragment key={node.path}>
           {isRoot && node.prev && <Divider className={classes.divider} />}
-          <Typography
-            variant="body2"
-            className={classNames(classes.link, classes.hover, {
-              [classes.header]: isRoot,
-              [classes.linkActive]: isLinkActive,
-              [classes.childWithInlineNodes]: hasInlineNodes,
-            })}
-            component={Link}
-            to={href}>
-            {node.data.title || node.name}
-          </Typography>
+          <Link to={href}>
+            <Typography
+              variant="body2"
+              className={classNames(classes.link, classes.hover, {
+                [classes.header]: isRoot,
+                [classes.linkActive]: isLinkActive,
+                [classes.childWithInlineNodes]: hasInlineNodes,
+              })}>
+              {node.data.title || node.name}
+            </Typography>
+          </Link>
           {hasInlineNodes && this.renderInlineNodes(inlineNodes)}
           <ul className={classes.ul}>
             {nodes.map(
@@ -233,17 +233,16 @@ export default class DocsSidebarList extends Component {
     }
 
     return (
-      <Typography
-        variant="body2"
-        className={classNames(classes.link, classes.hover, {
-          [classes.linkActive]: isLinkActive,
-          [classes.header]: isRoot,
-        })}
-        key={node.path}
-        component={Link}
-        to={href}>
-        {node.data.title}
-      </Typography>
+      <Link key={node.path} to={href}>
+        <Typography
+          variant="body2"
+          className={classNames(classes.link, classes.hover, {
+            [classes.linkActive]: isLinkActive,
+            [classes.header]: isRoot,
+          })}>
+          {node.data.title}
+        </Typography>
+      </Link>
     );
   };
 
@@ -266,24 +265,24 @@ export default class DocsSidebarList extends Component {
         {DOCS_MENU_ITEMS.map(item => (
           <Fragment key={item.label}>
             {item.hasChildren && <Divider />}
-            <ListItem
-              name={item.label}
-              button
-              onClick={this.handleSectionClick}
-              component={Link}
-              to={removeReadmeFromPath(item.path)}
-              classes={{ container: classes.listItem }}>
-              <ListItemIcon>
-                <item.icon />
-              </ListItemIcon>
-              <ListItemText primary={item.label} />
-              {item.hasChildren &&
-                (menuOpen && currentMenu === item.label ? (
-                  <ChevronUpIcon />
-                ) : (
-                  <ChevronDownIcon />
-                ))}
-            </ListItem>
+            <Link to={removeReadmeFromPath(item.path)}>
+              <ListItem
+                name={item.label}
+                button
+                onClick={this.handleSectionClick}
+                classes={{ container: classes.listItem }}>
+                <ListItemIcon>
+                  <item.icon />
+                </ListItemIcon>
+                <ListItemText primary={item.label} />
+                {item.hasChildren &&
+                  (menuOpen && currentMenu === item.label ? (
+                    <ChevronUpIcon />
+                  ) : (
+                    <ChevronDownIcon />
+                  ))}
+              </ListItem>
+            </Link>
             <Collapse
               in={item.hasChildren && menuOpen && currentMenu === item.label}>
               <div className={classes.collapse}>
diff --git a/ui/src/components/Dashboard/SidebarListItem.jsx b/ui/src/components/Dashboard/SidebarListItem.jsx
index a924d20f1..469780042 100644
--- a/ui/src/components/Dashboard/SidebarListItem.jsx
+++ b/ui/src/components/Dashboard/SidebarListItem.jsx
@@ -132,32 +132,33 @@ export default class SidebarListItem extends Component {
       children,
       rightIcon,
       staticContext,
+      to,
+      skipPrefetch,
       ...props
     } = this.props;
 
     return (
-      <ListItem
-        button
-        disableGutters
-        className={classes.listItem}
-        component={Link}
-        nav
-        isActive={this.isItemActive}
+      <Link
+        skipPrefetch={skipPrefetch}
+        to={to}
         activeClassName={classes.active}
-        {...props}>
-        {icon && (
-          <ListItemIcon classes={{ root: classes.icon }}>{icon}</ListItemIcon>
-        )}
-        <ListItemText
-          disableTypography
-          className={classes.text}
-          primary={children}
-        />
-        {rightIcon &&
-          cloneElement(rightIcon, {
-            className: classNames(classes.icon, rightIcon.props.className),
-          })}
-      </ListItem>
+        nav
+        isActive={this.isItemActive}>
+        <ListItem button disableGutters className={classes.listItem} {...props}>
+          {icon && (
+            <ListItemIcon classes={{ root: classes.icon }}>{icon}</ListItemIcon>
+          )}
+          <ListItemText
+            disableTypography
+            className={classes.text}
+            primary={children}
+          />
+          {rightIcon &&
+            cloneElement(rightIcon, {
+              className: classNames(classes.icon, rightIcon.props.className),
+            })}
+        </ListItem>
+      </Link>
     );
   }
 }
diff --git a/ui/src/components/Dashboard/index.jsx b/ui/src/components/Dashboard/index.jsx
index 721707401..e17e0ef2c 100644
--- a/ui/src/components/Dashboard/index.jsx
+++ b/ui/src/components/Dashboard/index.jsx
@@ -335,13 +335,13 @@ export default class Dashboard extends Component {
                 )}
               </IconButton>
             </Tooltip>
-            <Tooltip placement="bottom" title="Documentation">
-              <Link to={DOCS_PATH_PREFIX}>
+            <Link to={DOCS_PATH_PREFIX}>
+              <Tooltip placement="bottom" title="Documentation">
                 <IconButton>
                   <BookOpenPageVariantIcon className={classes.appIcon} />
                 </IconButton>
-              </Link>
-            </Tooltip>
+              </Tooltip>
+            </Link>
             {helpView && (
               <Tooltip placement="bottom" title="Page Information">
                 <IconButton onClick={this.handleHelpViewToggle}>
```

_Note: This will remove the warnings but the sidebar doesn't look like before. You might need to move some classnames to be on the parent Link. I didn't look into it but hopefully that should get you unblocked._

Reply:
Done with Dashboard! 
As far as I understand do I need to make the same changes with all `component={link} to={...}`?

Reply:
> As far as I understand do I need to make the same changes with all component={link} to={...}?

My current understanding is that yes, we should. Otherwise, there should be a warning in the console when the view that has `component={Link}` is being rendered.